shim-array.js

```diff
@@ -53,6 +53,17 @@ Array.unzip = function (table) {
 };
 
 function define(key, value) {
+
+    if (Array.prototype.hasOwnProperty(key)) {
+        if (
+            typeof console !== 'undefined' &&
+                typeof console.warn === 'function'
+        ) {
+            console.warn('Overrided Array.prototype.' + key, "Saved into _" + key);
+        }
+        define("_" + key, Object.getOwnPropertyDescriptor(Array.prototype, key).value);
+    }
```

Comment:
…but in general I’m not sure what purpose is served by setting aside the punched property. I don’t think anyone can rely on overridden properties being set aside, and in the cases where you care, it will be clearer to do so manually.

Reply:
done via ArrayFindPrototype